### PR TITLE
[Config Option] Added showLoader to VirtualTourPlugin

### PR DESCRIPTION
**Pull request by imjfdeviar**
**Merge request checklist**

-   [x] All lints and tests pass. If needed, new unit tests were added.
-   [ ] If needed, the [documentation](https://github.com/mistic100/Photo-Sphere-Viewer/tree/main/docs) has been updated.

#### `transitionOptions`

-   type: `object | function`
-   default: `{ speed: '20rpm', fadeIn: true, rotation: true, showLoader: true }`
-   updatable: no

Configuration of the transition between nodes. Can be a callback.

::: dialog "See details" "Virtual tour transitionOptions"

`transitionOptions` can be defined as a static object or a function called before switching to a new node.

The default behaviour is to rotate the view to face the direction of the link and perform a fade-in transition to the next node.

**If defined as an object, the type is:**

```ts
{
    /**
     * Speed or duration of the transition between nodes
     * @default '20rpm'
     */
    speed?: string | number;
    /**
     * show the loader while loading the new panorama
     * @default true
    */
    showLoader?: boolean;
    /**
     * Enable fade-in between nodes
     * @default true
     */
    fadeIn?: boolean;
    /**
     * Enable rotation in the direction of the next node
     * @default true
     */
    rotation?: boolean;
}
```

**If defined as a function, the signature is:**

```ts
(toNode: Node, fromNode?: Node, fromLink?: NodeLink) => ({
    speed?: string | number;
    /**
     * show the loader while loading the new panorama
     * @default true
    */
    showLoader?: boolean;
    fadeIn?: boolean;
    rotation?: boolean;
    /**
     * Define where to rotate the current panorama before switching to the next
     * if not defined it will use the link's position
     */
    rotateTo?: Position;
    /**
     * Define the new zoom level
     * if not defined it will keep the current zoom level
     */
    zoomTo?: number;
})
```

:::

### CODE REVIEW

**Comment by mistic100** [comment]
You can already call `setPanorama` with option `showLoader: false`, there is no need for an option.

Plus I don't like the fact it totally disables the component, it might be called in other contexts.

**Comment by imjfdeviar** [comment]
I get it, I didn't see the option in setPanorama, I like the way it is implemented but I was using some plugins: `VirtualTour` and it does not have the option in `VirtualTourTransitionOptions`.

Im using a custom loading with the events: `panorama-loaded` and `load-progress` and I dont want to see the loader in general (Because of that I added this option in general features to disable it from a general way). I'll update this PR updating transition options for the tour plugin.

BTW, I would like to contribute with this Repo, I would like to learn, with your experience, how to archive this goal? or where to add this feature?

File: packages/virtual-tour-plugin/src/VirtualTourPlugin.ts [hunk 1]
```diff
@@ -419,7 +421,10 @@ export class VirtualTourPlugin extends AbstractConfigurablePlugin<
                     throw utils.getAbortError();
                 }
 
-                this.viewer.loader.show();
+                if(transitionOptions.showLoader)
```
Comment (mistic100) [inline packages/virtual-tour-plugin/src/VirtualTourPlugin.ts:424]: please respect the code style

**Comment by mistic100** [comment]
Adding an option to VirtualTourPlugin is a good solution.

The documentation is in the same repo https://github.com/mistic100/Photo-Sphere-Viewer/blob/main/docs/plugins/virtual-tour.md you can update it yourself

**Review by mistic100 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**